Userland/Libraries/LibCore/System.cpp

```diff
@@ -1758,6 +1766,15 @@ ErrorOr<String> current_executable_path()
     auto ret = _NSGetExecutablePath(path, &size);
     if (ret != 0)
         return Error::from_errno(ENAMETOOLONG);
+#elif defined(AK_OS_HAIKU)
+    image_info info;
+    for (int32 cookie { 0 }; get_next_image_info(B_CURRENT_TEAM, &cookie, &info) == B_OK && info.type != B_APP_IMAGE;)
+        ;
+    if (info.type != B_APP_IMAGE)
+        return Error::from_string_view("current_executable_path() failed"sv);
+    if (sizeof(info.name) > sizeof(path))
+        return Error::from_errno(ENAMETOOLONG);
+    strcpy(path, info.name);
```

Comment:
`strncpy(path, info.name, sizeof(path) - 1)` to be extra "best practices"

Reply:
Haiku also has `strlcpy`, that could be used instead.

Reply:
I implemented the solution with strlcpy now, as I think it's the most failsafe option.
I hope I got that right.